### PR TITLE
Refacto / update FO tests pages for the js-doc

### PR DESCRIPTION
**Pull request by SD1982**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 1.7.8.x 
| Description?      | Refacto of the FO tests page for the js-doc
| Type?             | refacto
| Category?         | TE 
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | any
| How to test?      | Nothing to test 
| Possible impacts? | any.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/24163)

### CODE REVIEW

**Review by matks — DISMISSED** [review]
Hi @SD1982 I am sorry you have git conflicts. Can you please rebase?

File: tests/UI/pages/FO/contactUs/index.js [hunk 1]
```diff
@@ -35,8 +35,9 @@ class ContactUs extends FOBasePage {
 
   /**
    * Send message
-   * @param page
-   * @param contactUsData
+   * @param page {Page} Browser tab
+   * @param contactUsData {object} Object that contain the data to fill the form
+   * @param file
```
Comment (Progi1984) [inline tests/UI/pages/FO/contactUs/index.js]: What is the type of `file` ?

**Review by Progi1984 — COMMENTED** [review]
I think you should remove `String of the`. We knew already that the type is `String`.

File: tests/UI/pages/FO/contactUs/index.js (hunk 1 above)
Comment (SD1982) [reply tests/UI/pages/FO/contactUs/index.js]: Oh i forgot this one Thanks Progi1984 !

**Comment by SD1982** [comment]
> I think you should remove `String of the`. We knew already that the type is `String`.

It's modified now :+1:

File: tests/UI/pages/FO/cart/index.js [hunk 2]
```diff
@@ -143,8 +143,8 @@ class Cart extends FOBasePage {
 
   /**
    * Set promo code
-   * @param page
-   * @param code
+   * @param page {Page} Browser tab
+   * @param code {string} The promo code
```
Comment (boubkerbribri) [inline tests/UI/pages/FO/cart/index.js:147]:
```suggestion
   * @param code {string} The promo code to apply
```

File: tests/UI/pages/FO/checkout/index.js [hunk 3]
```diff
@@ -97,9 +96,9 @@ class Checkout extends FOBasePage {
 
   /**
    * Choose shipping method and add a comment
-   * @param page
-   * @param shippingMethod
-   * @param comment
+   * @param page {Page} Browser tab
+   * @param shippingMethod {string} String of the shipping method
+   * @param comment {string} String for the shipping comment
```
Comment (boubkerbribri) [inline tests/UI/pages/FO/checkout/index.js]:
```suggestion
   * @param shippingMethod {number} Position of the shipping method
   * @param comment {string} Comment to add after selecting a shipping method
```

File: tests/UI/pages/FO/checkout/index.js [hunk 4]
```diff
@@ -110,8 +109,8 @@ class Checkout extends FOBasePage {
 
   /**
    * Is shipping method exist
-   * @param page
-   * @param shippingMethod
+   * @param page {Page} Browser tab
+   * @param shippingMethod {string} String of the shipping method
```
Comment (boubkerbribri) [inline tests/UI/pages/FO/checkout/index.js]:
```suggestion
   * @param shippingMethod {number} Position of the shipping method
```

File: tests/UI/pages/FO/checkout/index.js [hunk 5]
```diff
@@ -207,17 +206,17 @@ class Checkout extends FOBasePage {
...
-    page.click(this.signInLink);
+    await page.click(this.signInLink);
   }
 
   /**
    * Login in FO
-   * @param page
-   * @param customer
+   * @param page {Page} Browser tab
+   * @param customer {object} The customer infos
```
Comment (boubkerbribri) [inline tests/UI/pages/FO/checkout/index.js]:
```suggestion
   * @param customer {object} Customer's information (email and password)
```

File: tests/UI/pages/FO/checkout/index.js [hunk 6]
```diff
@@ -295,8 +294,8 @@ class Checkout extends FOBasePage {
 
   /**
    * Set address
-   * @param page
-   * @param address
+   * @param page {Page} Browser tab
+   * @param address {object} The address infos
```
Comment (boubkerbribri) [inline tests/UI/pages/FO/checkout/index.js]:
```suggestion
   * @param address {object} Address's information to add
```

File: tests/UI/pages/FO/checkout/index.js [hunk 7]
```diff
@@ -312,8 +311,8 @@ class Checkout extends FOBasePage {
 
   /**
    * Fill personal information form and click on continue
-   * @param page
-   * @param customerData
+   * @param page {Page} Browser tab
+   * @param customerData {object} The customer data
```
Comment (boubkerbribri) [inline tests/UI/pages/FO/checkout/index.js]:
```suggestion
   * @param customerData {object} Guest Customer's information to fill on form
```

File: tests/UI/pages/FO/home/index.js [hunk 8]
```diff
@@ -236,9 +236,9 @@ class Home extends FOBasePage {
 
   /**
    * Select product color
-   * @param page
-   * @param id
-   * @param color
+   * @param page {Page} Browser tab
+   * @param id {number} ID of the color
+   * @param color {string} The color
```
Comment (boubkerbribri) [inline tests/UI/pages/FO/home/index.js]:
```suggestion
   * @param page {Page} Browser tab
   * @param id {number} Id of the current product
   * @param color {string} The color to select
```

File: tests/UI/pages/FO/home/index.js [hunk 9]
```diff
@@ -304,8 +304,8 @@ class Home extends FOBasePage {
 
   /**
    * Go to social sharing link
-   * @param page
-   * @param socialSharing
+   * @param page {Page} Browser tab
+   * @param socialSharing {string} The social link
```
Comment (boubkerbribri) [inline tests/UI/pages/FO/home/index.js]:
```suggestion
   * @param socialSharing {string} The social network name
```

File: tests/UI/pages/FO/home/index.js [hunk 10]
```diff
@@ -332,8 +332,8 @@ class Home extends FOBasePage {
 
   /**
    * Subscribe to the newsletter from the FO homepage
-   * @param {object} page
-   * @param {string} email
+   * @param page {Page} Browser tab
+   * @param email {string} The email
```
Comment (boubkerbribri) [inline tests/UI/pages/FO/home/index.js]:
```suggestion
   * @param email {string} Email to set on input
```

File: tests/UI/pages/FO/login/index.js [hunk 11]
```diff
@@ -22,8 +22,8 @@ class Login extends FOBasePage {
 
   /**
    * Login in FO
-   * @param page
-   * @param customer
+   * @param page {Page} Browser tab
+   * @param customer {object} Object of the customer infos (email, password, ....)
```
Comment (boubkerbribri) [inline tests/UI/pages/FO/login/index.js]:
```suggestion
   * @param customer {object} Customer's information (email and password)
```

File: tests/UI/pages/FO/myAccount/add.js [hunk 12]
```diff
@@ -39,8 +39,8 @@ class CreateAccount extends FOBasePage {
 
   /**
    * Create new customer account
-   * @param page
-   * @param customer
+   * @param page {Page} Browser tab
+   * @param customer {object} Object of the customer infos (email, password, ....)
```
Comment (boubkerbribri) [inline tests/UI/pages/FO/myAccount/add.js]:
```suggestion
   * @param customer {object} Customer's information (email and password)
```

File: tests/UI/pages/FO/myAccount/addAddress.js [hunk 13]
```diff
@@ -42,8 +42,8 @@ class AddAddress extends FOBasePage {
 
   /**
    * Fill address form and save
-   * @param page
-   * @param addressData
+   * @param page {Page} Browser tab
+   * @param addressData {object} Object for the address data (alias, firstname, lastname, company, address, city ...)
```
Comment (boubkerbribri) [inline tests/UI/pages/FO/myAccount/addAddress.js]:
```suggestion
   * @param addressData {object} Address's information to fill on form

File: tests/UI/pages/FO/myAccount/identity.js [hunk 14]
```diff
@@ -28,9 +28,9 @@ class AccountIdentity extends FOBasePage {
    */
   /**
    * Edit account information
-   * @param page
-   * @param oldPassword
-   * @param customer
+   * @param page {Page} Browser tab
+   * @param oldPassword {string} String for the old password
+   * @param customer {object} Object of the customer infos ( firstname, lastname, email, password)
```
Comment (boubkerbribri) [inline tests/UI/pages/FO/myAccount/identity.js]:
```suggestion
   * @param customer {object} Customer's information to fill form

File: tests/UI/pages/FO/passwordReminder/index.js [hunk 15]
```diff
@@ -22,8 +22,8 @@ class PasswordReminder extends FOBasePage {
 
   /**
    * Fill the reset password email form field and click "send reset link" button
-   * @param page
-   * @param email
+   * @param page {Page} Browser tab
+   * @param email {string} The email
```
Comment (boubkerbribri) [inline tests/UI/pages/FO/passwordReminder/index.js]:
```suggestion
   * @param email {string} Account's email to fill on input

File: tests/UI/pages/FO/product/index.js [hunk 16]
```diff
@@ -81,8 +81,8 @@ class Product extends FOBasePage {
 
   /**
    * Get product attributes from a Ul selector
-   * @param page
-   * @param ulSelector
+   * @param page {Page} Browser tab
+   * @param ulSelector {string} The selector
```
Comment (boubkerbribri) [inline tests/UI/pages/FO/product/index.js]:
```suggestion
   * @param ulSelector {string} Selector to locate the element

File: tests/UI/pages/FO/product/index.js [hunk 17]
```diff
@@ -202,9 +202,9 @@ class Product extends FOBasePage {
 
   /**
    * Select product combination
-   * @param page
-   * @param quantity
-   * @param combination
+   * @param page {Page} Browser tab
+   * @param quantity {number} Quantity of the product
+   * @param combination {object}  The product data
```
Comment (boubkerbribri) [inline tests/UI/pages/FO/product/index.js]:
```suggestion
   * @param quantity {number} Quantity of the product that customer wants
   * @param combination {object}  Product's combination data to select
```

File: tests/UI/pages/FO/product/index.js [hunk 18]
```diff
@@ -225,10 +225,10 @@ class Product extends FOBasePage {
...
   /**
    * Click on Add to cart button then on Proceed to checkout button in the modal
-   * @param page
-   * @param quantity
-   * @param combination
-   * @param proceedToCheckout
+   * @param page {Page} Browser tab
+   * @param quantity {number} Quantity of the product
+   * @param combination {object}  The product data
+   * @param proceedToCheckout {boolean} Boolean for proceed to checkout
```
Comment (boubkerbribri) [inline tests/UI/pages/FO/product/index.js]:
```suggestion
   * @param quantity {number} Quantity of the product that customer wants
   * @param combination {object}  Product's combination data to add to cart
   * @param proceedToCheckout {boolean} True to click on proceed to checkout button on modal
```

File: tests/UI/pages/FO/product/index.js [hunk 19]
```diff
@@ -250,8 +250,8 @@ class Product extends FOBasePage {
 
   /**
    * Go to social sharing link
-   * @param page
-   * @param socialSharing
+   * @param page {Page} Browser tab
+   * @param socialSharing {string} The social network
```
Comment (boubkerbribri) [inline tests/UI/pages/FO/product/index.js]:
```suggestion
   * @param socialSharing {string} Social network's name to get link from
```

File: tests/UI/pages/FO/product/index.js [hunk 20]
```diff
@@ -325,21 +325,19 @@ class Product extends FOBasePage {
...
-    return exist;
+    return await page.$(this.productSizeOption(size)) !== null;
   }
 
   /**
    * Is unavailable product color displayed
-   * @param page
-   * @param color
+   * @param page {Page} Browser tab
+   * @param color {string} The color
```
Comment (boubkerbribri) [inline tests/UI/pages/FO/product/index.js]:
```suggestion
   * @param color {string} Product's color to check
```

File: tests/UI/pages/FO/searchResults/index.js [hunk 21]
```diff
@@ -43,8 +43,8 @@ class SearchResults extends FOBasePage {
 
   /**
    * Click on Quick view Product
-   * @param page
-   * @param id, index of product in list of products
+   * @param page {Page} Browser tab
+   * @param id {number} Index of product in list of products
```
Comment (boubkerbribri) [inline tests/UI/pages/FO/searchResults/index.js]:
```suggestion
   * @param id {number} Index of product on the list
```

File: tests/UI/pages/FO/searchResults/index.js [hunk 22]
```diff
@@ -33,8 +33,8 @@ class SearchResults extends FOBasePage {
 
   /**
    * Go to the product page
-   * @param page
-   * @param id, product id
+   * @param page {Page} Browser tab
+   * @param id {number} Product id
```
Comment (boubkerbribri) [inline tests/UI/pages/FO/searchResults/index.js]:
```suggestion
   * @param id {number} Index of product on the list
```

File: tests/UI/pages/install/index.js [hunk 23]
```diff
@@ -199,8 +199,8 @@ class Install extends CommonPage {
   /**
    * Check if step installation is finished
    * @param page {Page} Browser tab
-   * @param step
-   * @param timeout
+   * @param step {string} The installation step
+   * @param timeout {number} Timeout duration
```
Comment (boubkerbribri) [inline tests/UI/pages/install/index.js]:
```suggestion
   * @param timeout {number} Time to wait for step to finish
```

**Review by Progi1984 — APPROVED** [review]

**Review by boubkerbribri — APPROVED** [review]

**Comment by Progi1984** [comment]
Thanks @SD1982

**Pull request merged**